### ISSUE TITLE
[autopilot] Calibration curves are identically zero for every arena-driven history — get_calibration_data ignores string outcomes and folds draws into losses

### ISSUE THREAD

**Issue by wdm0006** [helton-tech/elote#117]
## Problem

`History.get_calibration_data` is the only place in `History` that reads `Bout.outcome` **without** going through `Bout._normalized_outcome()`. It does a raw float comparison (`elote/arenas/base.py:673`):

```python
y_true.append(1.0 if bout.outcome == 1.0 else 0.0)
```

`LambdaArena.matchup()` — the primary way bouts get created — records its outcomes as the **strings** `"win"`, `"loss"` and `"tie"` (`elote/arenas/lambda_arena.py:150`, `:157`, `:164`). `"win" == 1.0` is `False`, so every bout an arena produces maps to `y_true = 0.0`.

Measured on `master` (b786851), 10 competitors of fixed distinct strength, 2000 `arena.matchup()` calls with Elo:

```
n bouts: 2000
sample outcomes: ['win', 'win', 'loss', 'loss', 'win']
sum(y_true) = 0.0 of 2000
prob_true: [0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]
prob_pred: [0.029 0.149 0.243 0.343 0.443 0.541 0.655 0.753 0.852 0.973]
```

Every bin's fraction-of-positives is zero, so `visualization.plot_calibration_curve` / `plot_calibration_comparison` draw a flat line along the x-axis for **every** rating system — the maximum-miscalibration signature — regardless of how good the predictions are. Both functions are part of the documented "diagnose prediction quality" journey, and `compute_calibration_data` is the sole consumer of `get_calibration_data` (`elote/visualization.py:228`).

The second half of the defect affects float-outcome histories too (the ones `evaluate_arena_with_dataset` / `evaluate_performance` build): a drawn bout has `outcome == 0.5`, which also falls into the `else` branch and is scored as a **full loss**. On `SyntheticDataset(60, 3000, seed=42, draw_probability=0.5, noise_std=200.0).time_split(0.3)` (900 eval bouts, 149 draws) with Elo:

| bin | n | mean predicted | `prob_true` today | `prob_true` with draws as 0.5 |
|---|---|---|---|---|
| 0.4–0.5 | 78 | 0.446 | 0.333 | **0.442** |
| 0.5–0.6 | 84 | 0.547 | 0.393 | **0.548** |
| 0.6–0.7 | 86 | 0.652 | 0.523 | **0.628** |
| 0.7–0.8 | 111 | 0.749 | 0.604 | **0.725** |
| 0.8–0.9 | 88 | 0.852 | 0.795 | **0.881** |

Elo is close to well-calibrated on this data, and the current code reports it as badly overconfident across the whole range.

Scoring a draw as 0.5 is not a judgment call here: every competitor's `expected_score` returns an *expected score* (win = 1, draw = 0.5, loss = 0), not P(win) — TrueSkill's returns `P(win) + P(draw)/2` explicitly. The calibration target has to be on the same scale as the quantity being calibrated.

Nothing catches this because `tests/test_calibration.py` builds its three fixture histories by hand with float `0.0`/`1.0` outcomes (`tests/test_calibration.py:48`, `:70`, `:89`) and never runs an arena, and `test_get_calibration_data` (`:94`) only asserts shape plus `all(y in [0.0, 1.0])`.

Separately, `get_calibration_data`'s `n_bins` parameter is accepted and never used — the binning happens entirely in `visualization.compute_calibration_data`.

## Proposed change

Make `get_calibration_data` use the same normalization every other `History` metric uses:

```python
label = bout._normalized_outcome()
if bout.predicted_outcome is None or label is None:
    skipped_bouts += 1
    continue
y_prob.append(bout.predicted_outcome)
y_true.append({"a": 1.0, "draw": 0.5, "b": 0.0}[label])
```

This fixes both halves at once and additionally makes unparseable outcomes *skip* (as `confusion_matrix` already does) instead of silently counting as losses. Update the docstring's `y_true` description accordingly.

Leave the `n_bins` parameter in place for backward compatibility but document that it is ignored (the binning lives in `visualization.compute_calibration_data`); `elote/visualization.py:228` can keep passing it.

`tests/test_calibration.py::test_get_calibration_data` asserts `all(y in [0.0, 1.0])` and must be widened to `[0.0, 0.5, 1.0]`. No other caller exists in `elote/`, `tests/`, `docs/` or `examples/`.

## Acceptance

- [ ] `get_calibration_data` resolves outcomes via `Bout._normalized_outcome()`, mapping `"a"` -> 1.0, `"draw"` -> 0.5, `"b"` -> 0.0, and skipping bouts whose outcome does not normalize (counting them in the existing `skipped_bouts` log).
- [ ] A regression test drives a real `LambdaArena` (string outcomes from `matchup()`), and asserts `sum(y_true) > 0` and that `compute_calibration_data` returns a `prob_true` that increases with `prob_pred`. This test fails on current `master` (every `prob_true` entry is exactly 0.0).
- [ ] A regression test builds a `History` containing a drawn bout and asserts the corresponding `y_true` entry is exactly 0.5.
- [ ] A regression test asserts a bout with an unrecognized outcome (e.g. `outcome="???"`) is skipped rather than emitted as 0.0, and that `len(y_true) == len(y_prob)` still holds.
- [ ] `tests/test_calibration.py::test_get_calibration_data` is updated to allow 0.5, and every other assertion in `tests/test_calibration.py` / `tests/test_visualization.py` still passes unchanged.
- [ ] The PR body states which mutation each new test catches — specifically, that the arena test is the one that fails if the raw `== 1.0` comparison is reinstated, and the draw test is the one that fails if draws are mapped to 0.0 while strings are handled.
- [ ] `get_calibration_data`'s docstring no longer claims `y_true` is "1.0 for wins, 0.0 for losses" and notes that `n_bins` is unused.
- [ ] `env -u VIRTUAL_ENV uv run --extra dev pytest -q --ignore=tests/test_examples.py` is green (280 passed / 6 skipped at this branch point, plus the new tests), and `make lint` passes.

## Risk

low — one method, one call site, no rating maths touched. It changes the numbers reported by the calibration plots, but in the direction of correctness: the current output is a constant zero for the library's own primary bout source. The only existing assertion that has to move is the `y in [0.0, 1.0]` shape check.
